ingenialink/servo.py

```diff
@@ -195,123 +188,1050 @@ def stop(self):
+            list_axis (list): List of all the axis in the dictionary.
+            subnode (int): Subnode to keep in the dictionary.
+
+        """
+        for axis in list_axis:
+            registers_category = axis.find('./Registers')
+            registers = registers_category.findall('./Register')
+            if subnode is not None and axis.attrib['subnode'] == str(subnode):
+                self.__update_single_axis_dict(registers_category, registers, subnode)
+                device.append(registers_category)
+                device.remove(axes_category)
+                break
+            for register in registers:
+                element_subnode = int(register.attrib['subnode'])
+                if (
+                        subnode in [None, element_subnode]
+                        and register.attrib.get('access') == 'rw'
+                ):
+                    self.__update_register_dict(register, element_subnode)
+                cleanup_register(register)
+
+    def __update_register_dict(self, register, subnode):
+        """Updates the register from a dictionary with the
+        storage parameters.
+
+        Args:
+            register (Element): Register element to be updated.
+            subnode (int): Target subnode.
+
+        Returns:
+
+        """
+        try:
+            storage = self.read(register.attrib['id'],
+                                subnode=subnode)
+            register.set('storage', str(storage))
+
+            # Update register object
+            reg = self._dictionary.registers(subnode)[register.attrib['id']]
+            reg.storage = storage
+            reg.storage_valid = 1
+        except BaseException as e:
+            logger.error("Exception during save_configuration, "
+                         "register %s: %s",
+                         str(register.attrib['id']), e)
+
+    def _set_state(self, state, subnode):
+        """Sets the state internally.
+
+        Args:
+            state (SERVO_STATE): Current servo state.
+            subnode (int): Subnode of the drive.
+
+        """
+        current_state = self.__state[subnode]
+        if current_state != state:
+            self.status[subnode] = state
+            for callback in self.__observers_servo_state:
+                callback(state, None, subnode)
+
+    def __read_coco_moco_register(self, register_coco, register_moco):
+        """Reads the COCO register and if it does not exist,
+        reads the MOCO register
+
+        Args:
+            register_coco (Register): COCO Register to be read.
+            register_moco (Register): MOCO Register to be read.
+
+        Returns:
+            int: Read value of the register.
+
+        """
+        try:
+            return self.read(register_coco, subnode=0)
+        except ILError:
+            pass
+
+        try:
+            return self.read(register_moco, subnode=1)
+        except ILError:
+            pass
```

Comment:
If not exist COCO nor MOCO register it returns None, is it the expected behavior?

Reply:
This cannot happen as the register_coco and register_moco registers are defined in the class.

Reply:
But it can file with a timeout or similar, in this case the error will not be raise and a None will be return

Reply:
Changed